server/text_generation_server/models/t5.py

```diff
@@ -29,6 +29,8 @@ def __init__(
         dtype: Optional[torch.dtype] = None,
         trust_remote_code: bool = False,
     ):
+        self.model_id = model_id
```

Comment:
I'm not a huge fan of this leak.

Couldn't we make a hash of the actual model object ? Not sure if torch has such primitives.

I'm more favorable of storing the model id within a global at launchtime (since there's hardly any world where the MODEL_ID needs to be changed at runtime) in order to avoid having this weird non global, but still kind of global value.
We've done this on other occasions to avoid bleeding parameters everywhere